### DISCUSSION TITLE
Makefile error in INET and Simu5G

### DISCUSSION THREAD

**Discussion by Queliewz** [omnetpp/omnetpp#1275] [Q&A] [answered]
![Screenshot 2024-11-30 112208](https://github.com/user-attachments/assets/d683246c-1bc4-48c1-a58c-d446271d0447)

I'm using OMNeT++ 6.0.3, INET 4.4.2 and Simu5G 1.2.2 built in the IDE. The makefile errors still appear and I don't know how to solve it. Please help!

**Comment by rhornig** [comment]
The output of the console where the actual error messages are dumped would be much more inbtereting.

**Reply by Queliewz** [reply]
Sorry, I delete all and install the lastest version of OMNeT++ and Simu5G with INET 4.5.2. Now, I'm encountering this @enum error.

![enum](https://github.com/user-attachments/assets/a18d493c-74ab-4e1a-9410-8752355161f5)

**Reply by rhornig** [reply]
There were changes regarding enum handling (in fact it was a bugfix) in omnetpp 6.1 but certain models (including inet) was using enums in a wrong way. Earlier versions of omnetpp missed these bugs in the model, but 6.1 is now complaining so models must be fixed also.

INET 4.5.4 already fixes this (anything earlier version of INET will not work).
It seems that Simu5G is also handling enums the wrong way. AFAIK this was fixed in Simu5G `dev` branch, but the official new release for Simu5G containing this fix is still pending.

So on short term, you should use omnetpp 6.0 to avoid this. 

NOTE: it seems that `opp_env` itself installs omnetpp 6.1 along with simu5g 1.2.2 which is definitely a bug in the `opp_env` database. If you try using `opp_env` you should force an earlier version of omnetpp to be installed (i.e. `opp_env install simu5g-latest omnetpp-6.0`) to avoid this issue.

**Reply by rhornig** [reply, marked as answer]
Simu5G 1.2.3 and 1.3.0 is out. Both versions are supposed to fix the above issue.

**Comment by Muhammaduazir69** [comment]
@Queliewz  Use opp_env for simu5g latest release. Check documentation for this on GitHub.
If you want manual configuration try to configure with omnet++ 5.6.2 and compatible INET
